src/role.ts

```diff
@@ -0,0 +1,360 @@
+   */
+  public grantRead(...grantees: IGrantee[]): void {
+    this.grant(['get', 'list', 'watch'], grantees);
+  }
+
+  /**
+   * Grant "get", "list", "watch", "create", "update", "patch", "delete", and
+   * "deletecollection" permissions for the grantee.
+   *
+   * @param grantees The resource(s) to apply to
+   */
+  public grantReadWrite(...grantees: IGrantee[]): void {
+    this.grant(['get', 'list', 'watch', 'create', 'update', 'patch', 'delete', 'deletecollection'], grantees);
+  }
+
+  /**
+   * Grant permission to perform a list of HTTP verbs on the grantee.
+   *
+   * @param grantees The resource(s) to apply to
+   * @see https://kubernetes.io/docs/reference/access-authn-authz/authorization/#determine-the-request-verb
+   */
+  public grant(verbs: string[], grantees: IGrantee[]): void {
+    for (const grantee of grantees) {
+      this.addRule({
+        apiGroups: grantee.apiGroups,
+        resources: grantee.resources,
+        resourceNames: grantee.resourceNames,
+        verbs,
+      });
+    }
+  }
+
+  /**
+   * List of rules included in this rule.
+   *
+   * Returns a copy. To add a rule, use `addRule()`.
+   */
+  public get rules() {
+    return [...this._rules];
+  }
+}
+
+/**
+ * A reference to any Role or ClusterRole.
+ */
+export interface IRole extends IResource {
+
+}
+
+/**
+ * Options for `Role`.
+ */
+export interface RoleProps extends RoleCommonProps {
+  /**
+   * The namespace of the resources this role's permissions should apply to
+   * (required).
+   */
+  readonly namespace: string;
+}
+
+/**
+ * Role is a namespaced, logical grouping of PolicyRules that can be referenced
+ * as a unit by a RoleBinding.
+ */
+export class Role extends RoleBase implements IRole {
+  /**
+   * @see base.Resource.apiObject
+   */
+  protected readonly apiObject: ApiObject;
+
+  constructor(scope: Construct, id: string, props: RoleProps) {
+    super(scope, id);
+
+    if (props.metadata?.namespace && props.metadata.namespace !== props.namespace) {
+      throw new Error('If `metadata.namespace` is passed as an option, its value must match the value of `namespace`.');
+    }
+
+    const metadata: ApiObjectMetadata = {
+      ...props.metadata,
+      namespace: props.namespace,
+    };
+
+    this.apiObject = new k8s.KubeRole(this, 'Resource', {
+      metadata,
+      rules: Lazy.any({ produce: () => this.synthesizeRules() }),
+    });
+  }
+
+  private synthesizeRules(): any {
+    return undefinedIfEmpty(this._rules.map(rule => rule._toKube()));
+  }
+
+  /**
+   * Create a RoleBinding that binds the permissions in this Role
+   * to a list of subjects, that will only apply this role's namespace.
+   * @param subjects a list of subjects to bind to
+   */
+  public bind(...subjects: ISubject[]): RoleBinding {
+    const binding = new RoleBinding(this, 'RoleBinding', {
+      namespace: this.metadata.namespace!,
+      role: this,
+    });
+    binding.addSubjects(...subjects);
+    return binding;
+  }
```

Comment:
Note: as implemented, you can't really call any of the bind methods (`bind`, `bindInNamespace`, `bindInCluster`) more than once on the same role because you'll get an error that there are two sibling constructs with the same name.

I don't think this is a blocker (since you can just create a RoleBinding directly and give it your own name) but thought it's worth calling out.